applications/api-cat/src/main/java/no/acat/service/ElasticsearchService.java

```diff
@@ -47,6 +47,13 @@ public ElasticsearchService(ObjectMapper mapper) {
         this.mapper = mapper;
     }
 
+    public ElasticsearchService(ObjectMapper mapper, Elasticsearch5Client elasticsearch, String clusterNodes, String clusterName) {
```

Comment:
Vi kan ikke ekstende kode til å introdusere slik test hooks.
Der må finnes en vei, hvordan man kan instansiere det via mockito.

En vei, for eksempel, kunne være å inject Envionment env, og da bruke 
localvar= env.getProperty("key.something")

Se her:
https://www.baeldung.com/properties-with-spring

Det kan sikker være flere alternativer til.

Reply:
Jeg er enig at  constractor bør bare ha de argumentene som er obligatoriske for å skape forekomsten av denne klassen. men normalt vil man sette kode for å initialisere en instansvariabel i en konstruktør. Java krever en constructor call  for hvert objekt som er opprettet, så dette er det ideelle utgangspunktet for å initialisere et objekts instansvariabler. jeg tror det er riktig på en objektorientert måte, fordi i konstruktøren er når objektet er opprettet, og det er når variabelen skal initialiseres. Jeg anbefaler å initialisere variabler i konstruktører. Det er derfor de eksisterer: for å sikre at objektene dine er konstruert (initialisert) riktig.